packages/devtools/src/components/ChatMessage/ChatMessage.tsx

```diff
@@ -22,38 +24,41 @@ interface ChatMessageProps {
 }
 
 const ChatMessage: FC<ChatMessageProps> = memo(
-  ({ content, streaming = false, feedback = false, sendDirection, value, onMessageAction }) => {
+  ({ content, streaming = false, feedback = true, sendDirection, value, onMessageAction }) => {
     const classes = useChatMessageStyles();
     const { deletedMessages, chat } = useChatStore();
     const isDeleted = deletedMessages[chat.id]?.some((m) => m.id === value.id);
     const labelId = `message-${value.id}`;
     const [isPopoverOpen, setIsPopoverOpen] = useState(false);
     const [openedByKeyboard, setOpenedByKeyboard] = useState(false);
     const [reactionSender, setReactionSender] = useState<MessageUser | undefined>();
+    const [isFeedbackDialogOpen, setIsFeedbackDialogOpen] = useState(false);
 
     const handleMessageKeyDown = useCallback(
       (event: React.KeyboardEvent) => {
         if (event.key === 'Enter' || event.key === ' ') {
           event.preventDefault();
-          setOpenedByKeyboard(true);
-          setIsPopoverOpen(true);
-          // Use setTimeout to ensure the Popover is rendered before trying to focus
-          setTimeout(() => {
-            const toolbar = document.querySelector(`[data-message-toolbar="${value.id}"] button`);
-            if (toolbar) {
-              (toolbar as HTMLButtonElement).focus();
-            }
-          }, 100);
+          if (!isFeedbackDialogOpen) {
+            setOpenedByKeyboard(true);
+            setIsPopoverOpen(true);
+            // Use setTimeout to ensure the Popover is rendered before trying to focus
+            setTimeout(() => {
+              const toolbar = document.querySelector(`[data-message-toolbar="${value.id}"] button`);
```

Comment:
Why not have the child pick up the focus when it's rendered in a useEffect

Reply:
(kinda a violation of separation of concerns here)

Reply:
This is on keydown (so an action is taken first) so `useEffect` doesn't apply here. Unless I'm misunderstanding your meaning?